modules/siddhi-extensions/string/src/main/java/org/wso2/siddhi/extension/string/EqualsIgnoreCaseFunctionExtension.java

```diff
@@ -0,0 +1,94 @@
+import org.wso2.siddhi.core.executor.function.FunctionExecutor;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.ExecutionPlanValidationException;
+
+/**
+ * equalsIgnoreCase(string, compareTo)
+ * Compares two strings lexicographically.
+ * Accept Type(s): (STRING,STRING)
+ * Return Type(s): BOOL
+ */
+public class EqualsIgnoreCaseFunctionExtension extends FunctionExecutor {
+    Attribute.Type returnType = Attribute.Type.BOOL;
+
+    @Override
+    public Attribute.Type getReturnType() {
+        // TODO Auto-generated method stub
+        return returnType;
+    }
+
+    @Override
+    public void start() {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public void stop() {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    public Object[] currentState() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public void restoreState(Object[] state) {
+        // TODO Auto-generated method stub
+
+    }
+
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+        // TODO Auto-generated method stub
+        if (attributeExpressionExecutors.length != 2) {
+            throw new ExecutionPlanValidationException(
+                    "Invalid no of arguments passed to str:equalsIgnoreCase() function, " + "required 2, but found "
+                            + attributeExpressionExecutors.length);
+        }
+        if (attributeExpressionExecutors[0].getReturnType() != Attribute.Type.STRING) {
+            throw new ExecutionPlanValidationException(
+                    "Invalid parameter type found for the first argument of str:equalsIgnoreCase() function, "
+                            + "required " + Attribute.Type.STRING + ", but found "
+                            + attributeExpressionExecutors[0].getReturnType().toString());
+        }
+        if (attributeExpressionExecutors[1].getReturnType() != Attribute.Type.STRING) {
+            throw new ExecutionPlanValidationException(
+                    "Invalid parameter type found for the second argument of str:equalsIgnoreCase()) function, "
+                            + "required " + Attribute.Type.STRING + ", but found "
+                            + attributeExpressionExecutors[1].getReturnType().toString());
+        }
+
+    }
+
+    @Override
+    protected Object execute(Object[] data) {
+        // TODO Auto-generated method stub
+        if (data[0] == null) {
+            throw new ExecutionPlanRuntimeException(
+                    "Invalid input given to str:equalsIgnoreCase() function. First argument cannot be null");
+        }
+        if (data[1] == null) {
+            throw new ExecutionPlanRuntimeException(
+                    "Invalid input given to str:equalsIgnoreCase() function. Second argument cannot be null");
+        }
```

Comment:
Checking nullity of source is enough in my opinion to avoid NPE. If compareTo is null equalsIgnoreCase anyway return false. @suhothayan WDYT?